packages/vike-server/src/plugin/plugins/devServerPlugin.ts

```diff
@@ -130,24 +158,82 @@ export function devServerPlugin(): Plugin {
     const url = new URL(req.url, 'http://example.com')
     return url.pathname === VITE_HMR_PATH
   }
+
+  function isImported(
+    modules: EnvironmentModuleNode[]
+  ): { type: 'entry' | '+middleware'; module: EnvironmentModuleNode } | undefined {
+    const modulesSet = new Set(modules)
+    for (const module of modulesSet.values()) {
+      if (module.file === resolvedEntryId)
+        return {
+          type: 'entry',
+          module
+        }
+      if (module.file?.match(/\+middleware\.[mc]?[jt]sx?$/))
+        return {
+          type: '+middleware',
+          module
+        }
+      // biome-ignore lint/complexity/noForEach: <explanation>
+      module.importers.forEach((importer) => modulesSet.add(importer))
+    }
+  }
+
+  function patchViteServer(vite: ViteDevServer) {
```

Comment:
I think I added this to be compatible with the vite cli, because vite cli calls these three on `vite dev`, even if we set `middlewaremode: true` in a plugin config hook.

Maybe these aren't needed anymore with the new vike cli, if the vike cli detects `middlewaremode` it can skip calling `vite.listen`, etc.. https://github.com/vikejs/vike/blob/main/vike/node/cli/entry.ts#L28.